File: app/src/main/java/rojoxpress/slideexample/MainActivity.java
package rojoxpress.slideexample;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.SwitchCompat;
import android.widget.CompoundButton;
import android.widget.TextView;
import android.widget.Toast;

import com.rojoxpress.slidebutton.SliderButton;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final TextView textView = (TextView) findViewById(R.id.progress);
        final SliderButton slideButton = (SliderButton) findViewById(R.id.slide_button);
        SwitchCompat switchCompat = (SwitchCompat) findViewById(R.id.switch_);

        slideButton.setSlideButtonListener(new SliderButton.SlideButtonListener() {
            @Override
            public void onSlide() {
                Toast.makeText(MainActivity.this,"UNLOCKED",Toast.LENGTH_SHORT).show();
            }
        });

        slideButton.setOnSlideChangeListener(new SliderButton.OnSlideChangeListener() {
            @Override
            public void onSlideChange(float position) {
                textView.setText("Progress: "+position);
            }
        });

        switchCompat.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
                slideButton.setEnabled(b);
            }
        });
    }
}

File: app/src/main/java/rojoxpress/slideexample/MainActivity.kt
package rojoxpress.slideexample

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Toast
import com.rojoxpress.slidebutton.SliderButton
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val textView = progress
        val slideButton = slide_button
        val switchCompat = switch_

        slideButton.setSlideButtonListener(object : SliderButton.OnSlideCompleteListener {
            override fun onSlideComplete() {
                Toast.makeText(this@MainActivity, "UNLOCKED", Toast.LENGTH_SHORT).show()
            }
        })

        slideButton.setOnSlideChangeListener(object : SliderButton.OnSlideChangeListener {
            override fun onSlideChange(position: Float) {
                textView.text = "Progress: $position"
            }
        })

        switchCompat.setOnCheckedChangeListener { compoundButton, b -> slideButton.isEnabled = b }
    }
}
